packages/flutter/lib/src/material/input.dart

```diff
@@ -134,19 +128,14 @@ class _InputState extends State<Input> {
   }
 
   void _handleTextUpdated() {
-    if (_value != _editableString.text) {
-      setState(() {
-        _value = _editableString.text;
-      });
-      if (config.onChanged != null)
-        config.onChanged(_value);
-    }
+    if (config.onChanged != null)
+      config.onChanged(_editableString.asInputValue());
```

Comment:
It would probably be cleaner to have editable string have an InputValue instead of ripping the fields out and putting them back together.